router/router.go

```diff
@@ -1275,6 +1279,7 @@ func (worker *workerT) handleJobForPrevFailedUser(job *jobsdb.JobT, parameters J
 			Parameters:    []byte(`{}`),
 			WorkspaceId:   job.WorkspaceId,
 		}
+		job.EventPayload = nil // offload job payload from memory before adding it to the response queue
```

Comment:
Any reason why are we setting the payload to nil ?

Reply:
We would need to job payload in case of aborted jobs to back them up in proc_error DB

Reply:
Yes, it is a first (poor man's) attempt to limit what we keep in memory on router.
As it seems, it is not enough that we limit the pickup size based on payload, since we have a lot of buffers where we are retaining jobs in memory:

1. We have `X` routers picking up jobs based on payload (100MB limit)
2. We have `X*64` worker.channels buffering **20 jobs** before processing them
3. We have a `responseQ` buffering **1000 jobs** before updating their status

So in the worst case scenario, assuming that we have **1MB** mean payload size we need 
```
X*100MB + X*64*20*1MB + 1000*1MB
```
For 10 different destination types this translates to 1GB+13GB+1GB=**14GB**! 
I was just trying to shave off 1 GB :/, but now that I am looking at the actual numbers, will revert this change and add another task to introduce payload size limits on the buffers too! 
@lvrach @chandumlg @Sidddddarth @saurav-malani

Reply:
https://www.notion.so/rudderstacks/8aac9087df644365acdf64e28e290153?v=6e06b0a5ade24f0aa5ffe05dc2972e84&p=364496e2ada346bea791033bfbf0965e

Reply:
> I was just trying to shave off 10 GB :/, but now that I am looking at the actual numbers, will revert this change and add another task to introduce payload size limits on the buffers too!

That is a good idea, keep tracking the total amount of bytes currently in the router (across all workers) would be really helpful. You can fetch based on the remaining payload budget.

A separate PR would be better to track this.